wallet/udb/upgrades.go

```diff
@@ -487,6 +496,108 @@ func slip0044CoinTypeUpgrade(tx walletdb.ReadWriteTx, publicPassphrase []byte) e
 	return unifiedDBMetadata{}.putVersion(metadataBucket, newVersion)
 }
 
+func hasExpiryUpgrade(tx walletdb.ReadWriteTx, publicPassphrase []byte) error {
+	const oldVersion = 7
+	const newVersion = 8
+	metadataBucket := tx.ReadWriteBucket(unifiedDBMetadata{}.rootBucketKey())
+	txmgrBucket := tx.ReadWriteBucket(wtxmgrBucketKey)
+
+	// Assert this function is only called on version 7 databases.
+	dbVersion, err := unifiedDBMetadata{}.getVersion(metadataBucket)
+	if err != nil {
+		return err
+	}
+	if dbVersion != oldVersion {
+		const str = "hasExpiryUpgrade inappropriately called"
+		return apperrors.E{ErrorCode: apperrors.ErrUpgrade, Description: str, Err: nil}
+	}
+
+	// Iterate through all mined credits and update stored serializations
+	creditsBucket := txmgrBucket.NestedReadWriteBucket(bucketCredits)
+	cursor := creditsBucket.ReadWriteCursor()
+	creditsKV := map[string][]byte{}
+	for k, v := cursor.First(); v != nil; k, v = cursor.Next() {
+		encodedK := hex.EncodeToString(k)
+		creditsKV[encodedK] = v
```

Comment:
Rather than saving all keys and values in the bucket in memory, check here whether the value must be rewritten and only add those that must be.  Save the modified copy of the value to this map, and then the loop below can just put every key and value in the map.